### PR TITLE
Simply 3945/update react beautiful dnd

### PR DESCRIPTION
**Pull request by oliviawongnyc**
## Description

- Updated the react-beautiful-dnd library from v2.3.1 to v11.0.2, and made necessary changes to the code and tests.
- The drag and drop functionality is used in the List Manager and Lane Manager.

## Motivation and Context

- The react-beautiful-dnd library was very outdated, and subsequent versions included performance and UX improvements.
- The drag and drop functionality is a part of the List Manager, which is the tool I am working on improving overall with the refactor.

## How Has This Been Tested?

- I have tested this update manually, on local host.
- I have updated unit tests, and all unit tests are passing.
- I **did not update** the skipped tests in `CustomListBuilder-test`. This is because enzyme is not well-suited to test user interactions (ie, dragging and dropping), nor stateful functional components with hooks. My next piece of work will be to bring in React Testing Library for these tests.

## Checklist:

- [x] I have updated the documentation accordingly.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/components/CustomListEntries.tsx [hunk 1]
```diff
@@ -13,32 +13,32 @@ export interface Entry extends BookData {
...
   nextPageUrl?: string;
   opdsFeedUrl?: string;
+  searchResults?: CollectionData;
   deleteAll: () => void;
   deleteEntry?: (id: string) => void;
   loadMoreEntries: (url: string) => Promise<CollectionData>;
-  setDraggingFrom: (dragging: string | null) => void;
   setLoadedMoreEntries: (clicked: boolean) => void;
+  // Including for test purposes
+  children?: any;
```
Comment (EdwinGuzman) [inline src/components/CustomListEntries.tsx:27]: Do you still need this?

File: src/components/Lane.tsx [hunk 2]
```diff
@@ -58,7 +58,7 @@ export default class Lane extends React.Component<LaneProps, LaneState> {
             (active ? "active" : "")
           }
           ref={provided && provided.innerRef}
-          style={provided && provided.draggableStyle}
+          {...(provided ? provided.draggableProps : {})}
           {...(provided ? provided.dragHandleProps : {})}
```
Comment (EdwinGuzman) [inline src/components/Lane.tsx:62]: Can you merge this into one statement?
```
{...(provided ? {...provided.draggableProps, ...provided.dragHandleProps } : {})}
```
Does that work?

File: src/components/LaneCustomListsEditor.tsx [hunk 3]
```diff
@@ -6,28 +6,18 @@ import TrashIcon from "./icons/TrashIcon";
...
-
 /** Drag and drop interface for adding custom lists to a lane. */
 export default class LaneCustomListsEditor extends React.Component<
-  LaneCustomListsEditorProps,
-  LaneCustomListsEditorState
+  LaneCustomListsEditorProps
 > {
   constructor(props) {
     super(props);
-    this.state = {
```
Comment (EdwinGuzman) [inline src/components/LaneCustomListsEditor.tsx:27]: Sweet

File: src/components/__tests__/CustomListBookCard-test.tsx [hunk 4]
```diff
@@ -15,28 +17,66 @@ export interface Entry extends BookData {
...
-  const searchResult: Entry = {
-    id: "1",
-    title: "result 1",
-    authors: ["author 1"],
-    url: "/some/url1",
-    language: "eng",
-    raw: {
-      $: { "schema:additionalType": { value: "http://schema.org/EBook" } },
-    },
+  const searchResultsData = {
```
Comment (EdwinGuzman) [inline src/components/__tests__/CustomListBookCard-test.tsx:43]: If you can give this a type, that'd be great. If not, it'd be good for a later PR.

**Review by EdwinGuzman — APPROVED** [review]
This works well and glad the big jump was manageable. I get the following warning but I don't think it's something you did wrong but rather how the lists are contained in a div that has a scrollbar, which was there before. I don't think it's a dealbreaker for this PR but it'd be good to get a design update so the two lists aren't scrollable, perhaps?

Screen Shot 2022-01-13 at 11 15 23 AM

File: src/components/Lane.tsx (hunk 2 above)
Comment (oliviawongnyc) [reply src/components/Lane.tsx:62]: Works! Thanks!

File: src/components/CustomListEntries.tsx (hunk 1 above)
Comment (oliviawongnyc) [reply src/components/CustomListEntries.tsx:27]: From what I can tell, yes. At least with Enzyme.

**Pull request merged**